scripts/report_functions.R

```diff
@@ -25,6 +25,56 @@
 # adapted from
 # https://github.com/rstudio/bookdown/blob/master/R/utils.R
 
+## Wrapper function to run a specific Rmd script
+## which does the computation and generates an HTML report with code included
+render2HTML <- function(reportFile,
+                            outFile,
+                            outDir,
+                            finalReportDir,
+                            report.params=NULL,
+                            self.contained=TRUE,
+                            logFile = NULL)
+{
+  
+  output_format = rmarkdown::all_output_formats(reportFile, 'UTF-8')
+  
+  if(!dir.exists(finalReportDir)) dir.create(finalReportDir, recursive = TRUE)
+  
+  if(is.null(report.params)) report.params <- list()
```

Comment:
This seems unnecessary.  Couldn't line 34 default to an empty list instead of NULL?